design/select/index.jsx

```diff
@@ -0,0 +1,190 @@
+import React, { useEffect, useState, useRef } from 'react'
+import classNames from 'classnames'
+
+import classNameHelpers from '../class-name-helpers.css'
+import styles from './styles.css'
+
+// TODO: Taken from @oacore/design
+const generateId = () =>
+  Math.random()
+    .toString(36)
+    .substr(2, 9)
+
+const KEYS = {
+  ESC: 27,
+  ENTER: 13,
+  LEFT: 37,
+  UP: 38,
+  RIGHT: 39,
+  DOWN: 40,
+}
+
+// TODO: Consider converting it to class component
+const Select = ({
+  onSelectionChange,
+  value,
+  options,
+  className,
+  tag: Tag = 'div',
+  id = generateId(),
+}) => {
+  const inputRef = useRef(null)
+  const [searchTerm, setSearchTerm] = useState('')
+  const [showSuggestions, toggleShowSuggestions] = useState(false)
+  const [activeSuggestion, setActiveSuggestion] = useState({})
+  const [loading, setLoading] = useState(false)
+  const [suggestions, setSuggestions] = useState([])
+
+  const handleKeyDown = event => {
+    let pos = suggestions.findIndex(s => s.id === activeSuggestion.id) || 0
+
+    switch (event.which) {
+      case KEYS.ESC:
+        inputRef.current.blur()
+        toggleShowSuggestions(false)
+        break
+
+      case KEYS.RIGHT:
+        inputRef.current.blur()
+        onSelectionChange(activeSuggestion)
+        break
```

Comment:
As a user I must be able to navigate my input in both directions. This disables such functionality.

Reply:
What do you mean?

Arrows UP/Down are for navigation. Right arrow and Enter is for selecting option.

Reply:
Only Enter should do it. Right arrow should navigate the input.